src/coreclr/tryrun.cmake

```diff
@@ -73,6 +72,26 @@ if(TARGET_ARCH_NAME MATCHES "^(armel|arm|arm64|x86)$" OR FREEBSD)
     set_cache_value(UNGETC_NOT_RETURN_EOF 0)
     set_cache_value(HAVE_COMPATIBLE_ILOGBNAN 1)
     set_cache_value(HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE 0)
+  elseif(ILLUMOS)
+    set_cache_value(GETPWUID_R_SETS_ERRNO_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_ACOS_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_ASIN_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_ATAN2_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_POW_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_ILOGBNAN_EXITCODE 0)
+    set_cache_value(HAVE_COMPATIBLE_LOG10_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_LOG_EXITCODE 1)
+    set_cache_value(HAVE_LARGE_SNPRINTF_SUPPORT 1)
```

Comment:
It seems these three should have the `_EXITCODE` at the end of the symbol name too, shouldn't they?

Reply:
That's correct. also fixed the FreeBSD case above.

Reply:
@janvorli, for some reason these three and the FreeBSD one also work with without `_EXITCODE` (I am not sure how, since all the `runnable` cases must have EXITCODE as the common knowledge dictates).

However, when I changed it to _EXITCODE, I forgot to invert the values, as the outcome is oppose to what the value represent. Caught it in the CI run on a public server.

Will fix it in a follow up PR.

Reply:
Fixed as part of #37824.